gateway/blocks_gateway.go

```diff
@@ -223,38 +231,208 @@ func (api *BlocksGateway) Head(ctx context.Context, path ImmutablePath) (Content
 	return md, fileNode, nil
 }
 
-func (api *BlocksGateway) GetCAR(ctx context.Context, path ImmutablePath) (ContentPathMetadata, io.ReadCloser, <-chan error, error) {
-	// Same go-car settings as dag.export command
-	store := dagStore{api: api, ctx: ctx}
+func (api *BlocksGateway) GetCAR(ctx context.Context, p ImmutablePath, params CarParams) (io.ReadCloser, <-chan error, error) {
+	contentPathStr := p.String()
+	if contentPathStr[0:6] != "/ipfs/" {
+		return nil, nil, fmt.Errorf("path does not have /ipfs/ prefix")
+	}
+	firstSegment, _, _ := strings.Cut(contentPathStr[6:], "/")
+	rootCid, err := cid.Decode(firstSegment)
+	if err != nil {
+		return nil, nil, err
+	}
 
-	// TODO: When switching to exposing path blocks we'll want to add these as well
-	roots, lastSeg, err := api.getPathRoots(ctx, path)
+	r, w := io.Pipe()
```

Comment:
I know this isn't new but this is meh, I think this should be refactored as `WriteCar` and take an `io.Writer` as input.

Reply:
If we do my `WriteCar` this wouldn't need a chan error anymore.
You would synchronously write to the reader and simply return an error. This code wouldn't spawn a background goroutine.
And in the current code we can use `.CloseWithError` on the pipe instead of `<-chan error`.

Reply:
@Jorropo yeah, I already changed that.

Reply:
I still think we should *consider* removing the go-routine and writing synchronously.